common/djangoapps/student/migrations/0011_auto_20170214_1200.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+def add_email_uniqueness_constraint(apps, schema_editor):
+    # Do we already have an email uniqueness constraint?
+    cursor = schema_editor.connection.cursor()
+    cursor.execute("""
+        select * from information_schema.table_constraints
+        where constraint_schema = 'edxapp'
+        and table_name = 'auth_user'
+        and constraint_name = 'email'
+        and constraint_type = 'UNIQUE'
+    """)
+    has_constraint = bool(list(cursor.fetchall()))
+    if not has_constraint:
+        # We don't have a constraint: add it.
+        schema_editor.execute("create unique index email on auth_user (email)")
```

Comment:
Can we conditionally push on an AlterField operation that adds uniqueness and let Django figure out the syntax for us?  Or do we want to do this to control the name of the index.